ui/src/app/subnets/views/SubnetsList/SubnetsTable/hooks.ts

```diff
@@ -41,9 +48,71 @@ export const useSubnetsTable = (
 
   useEffect(() => {
     if (loaded) {
-      setRows(getTableData({ fabrics, vlans, subnets, spaces }, groupBy));
+      setState({
+        data: getTableData({ fabrics, vlans, subnets, spaces }, groupBy),
+        loaded: true,
+      });
+    }
+  }, [loaded, fabrics, vlans, subnets, spaces, groupBy]);
+
+  return state;
+};
+
+export const useSubnetsTableSearch = (
+  subnetsTable: { data: SubnetsTableRow[]; loaded: boolean },
+  searchText: string
+): UseSubnetsTable => {
+  const [state, setState] = useState<UseSubnetsTable>({
+    data: [],
+    loaded: false,
+  });
+
+  useEffect(() => {
+    if (subnetsTable.loaded) {
+      setState({
+        data: filterSubnetsBySearchText(subnetsTable.data, searchText),
+        loaded: true,
+      });
     }
-  }, [loaded, fabrics, vlans, subnets, spaces, setRows, groupBy]);
+  }, [subnetsTable, searchText]);
 
-  return { rows };
+  return state;
 };
+
+export function usePagination<D>(
+  data: Array<D>,
+  itemsPerPage = SUBNETS_TABLE_ITEMS_PER_PAGE
+): {
```

Comment:
We could probably upstream this to `react-components` since it's pretty generic already, maybe include an arg for initial page (instead of it always being 0) and we're good to go!